WebApp/app.go

```diff
@@ -92,6 +96,101 @@ func simplifyImage(c *gin.Context) {
 	c.Header("Content-Disposition", `attachment; filename="simplified-image.png"`)
 }
 
+func buildLevel(c *gin.Context) {
+	c.SetAccepted("multipart/form-data")
+
+	fileh, err := c.FormFile("image")
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"errorMessage": `File not found, be sure it is included under the "image" key of your form`})
+		return
+	}
+
+	file, err := fileh.Open()
+	if err != nil {
+		panic(err)
+	}
+
+	contentType := fileh.Header.Get("Content-Type")
+	var img image.Image
+	switch contentType {
+	case "image/png":
+		img, err = png.Decode(file)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"errorMessage": "Could not decode your PNG image."})
+			return
+		}
+	case "image/jpeg":
+		img, err = jpeg.Decode(file)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"errorMessage": "Could not decode your JPEG image."})
+			return
+		}
+	default:
+		c.JSON(http.StatusBadRequest, gin.H{"errorMessage": "Only JPEG and PNG images are accepted."})
+		return
+	}
+
+	img, err = processing.ResizeImage(img)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"errorMessage": err.Error()})
+		return
+	}
+
+	newImg, regionCount := processing.SimplifyImage(img)
+
+	regionMap := processing.BuildRegionMap(newImg)
+
+	data := make([]regionData, regionCount)
+
+	for i := 0; i < regionCount; i++ {
+		region := regionMap.GetRegion(processing.RegionIndex(i))
+
+		minX, minY := processing.FindRegionPosition(region)
+		regionColor := processing.GetColorOfRegion(region, newImg)
+
+		regionImage := image.NewNRGBA(region.GetBounds())
+
+		for j := 0; j < len(region); j++ {
+			regionImage.Set(int(region[j].X), int(region[j].Y), processing.GetColorOfRegion(region, newImg))
+		}
+
+		buf := new(bytes.Buffer)
+		if err := png.Encode(buf, regionImage); err != nil {
+			panic(err)
+		}
+		base64Region := base64.StdEncoding.EncodeToString(buf.Bytes())
+
+		r := regionData{i, regionColor, minX, minY, base64Region}
+		data[i] = r
+	}
+
+	d, err := json.Marshal(data)
+	if err != nil {
+		panic(err)
+	}
+	log.Print(string(d))
+
+	listenerHub.NotifyListeners(ListenerMessage{
+		Type: "build-level",
+		Attachments: []AttachedFile{
+			{
+				Name:          "level.json",
+				ContentType:   "application/json",
+				Base64Content: string(d),
+			},
+		},
+		Timestamp: time.Now().Format(time.RFC3339),
+	})
+}
```

Comment:
We're sending the JSON file to the bot, but we also need to send it back in the HTTP response so that the game can use it.
```suggestion
	c.Data(http.StatusOK, "application/json", d)
}
```
I think this will work, you can verify that it does by running this command in a terminal while the web app is running:
```ps
curl.exe -X POST -F image=@./image.png -o './level.json' http://localhost:8080/api/build-level
```
(replace `image.png` with the file name of your test image)